diff --git a/server/token.c b/server/token.c
index ec2d753..d5df928 100644
--- a/server/token.c
+++ b/server/token.c
@@ -213,7 +213,7 @@ void init_user_sid(void)
     size_t n;
     FILE *f;

-    f = fopen( "/etc/machine-id", "r" );
+    f = fopen( "/data/data/com.termux/files/usr/etc/machine-id", "r" );
     if (!f)
     {
         fprintf( stderr, "Failed to open /etc/machine-id, error %s.\n", strerror(>

