diff --git a/dlls/ntdll/unix/fsync.c b/dlls/ntdll/unix/fsync.c
index c3da44e..13cb521 100644
--- a/dlls/ntdll/unix/fsync.c
+++ b/dlls/ntdll/unix/fsync.c
@@ -165,12 +165,14 @@ static inline int futex_wake( int *addr, int val )
 
 int do_fsync(void)
 {
-#ifdef __linux__
+#if defined(__linux__) && !defined(__ANDROID__)
     static int do_fsync_cached = -1;
 
     if (do_fsync_cached == -1)
     {
+        #if defined(__linux__) && !defined(__ANDROID__)
         syscall( __NR_futex_waitv, NULL, 0, 0, NULL, 0 );
+        #endif
         do_fsync_cached = getenv("WINEFSYNC") && atoi(getenv("WINEFSYNC")) && errno != ENOSYS;
     }
 
@@ -570,10 +572,10 @@ void fsync_init(void)
         ERR("Cannot stat %s\n", config_dir);
 
     if (st.st_ino != (unsigned long)st.st_ino)
-        sprintf( shm_name, "/wine-%lx%08lx-fsync", (unsigned long)((unsigned long long)st.st_ino >> 32), (unsigned long)st.st_ino );
+        sprintf( shm_name, "/data/data/com.termux/files/usr/tmp/wine-%lx%08lx-fsync", (unsigned long)((unsigned long long)st.st_ino >> 32), (unsigned long)st.st_ino );
     else
-        sprintf( shm_name, "/wine-%lx-fsync", (unsigned long)st.st_ino );
-
+        sprintf( shm_name, "/data/data/com.termux/files/usr/tmp/wine-%lx-fsync", (unsigned long)st.st_ino );
+    #if defined(__linux__) && !defined(__ANDROID__)
     if ((shm_fd = shm_open( shm_name, O_RDWR, 0644 )) == -1)
     {
         /* probably the server isn't running with WINEFSYNC, tell the user and bail */
@@ -586,6 +588,7 @@ void fsync_init(void)
 
     current_pid = GetCurrentProcessId();
     assert(current_pid);
+    #endif
 }
 
 NTSTATUS fsync_create_semaphore( HANDLE *handle, ACCESS_MASK access,
diff --git a/dlls/ntdll/unix/loader.c b/dlls/ntdll/unix/loader.c
index e4f110d..3a0abe7 100644
--- a/dlls/ntdll/unix/loader.c
+++ b/dlls/ntdll/unix/loader.c
@@ -2255,7 +2255,9 @@ static void start_main_thread(void)
     dbg_init();
     startup_info_size = server_init_process();
     hacks_init();
+#if defined(__linux__) && !defined(__ANDROID__)
     fsync_init();
+#endif
     esync_init();
     virtual_map_user_shared_data();
     init_cpu_info();
diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index 9236974..b21b577 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -408,8 +408,9 @@ static void kernel_writewatch_softdirty_init(void)
 static void kernel_writewatch_init(void)
 {
     struct uffdio_api uffdio_api;
-
+#if defined __linux__ && !defined __ANDROID__
     uffd_fd = syscall( __NR_userfaultfd, O_CLOEXEC | O_NONBLOCK | UFFD_USER_MODE_ONLY );
+#endif
     if (uffd_fd == -1)
     {
         kernel_writewatch_softdirty_init();
@@ -6271,8 +6272,9 @@ NTSTATUS get_memory_fex_stats_shm( HANDLE process, LPCVOID addr, MEMORY_FEX_STAT
 
         oflag |= O_CREAT | O_TRUNC;
     }
-
+#if defined(__linux__) && !defined(__ANDROID__)
     fd = shm_open( buf, oflag, S_IRWXU | S_IRWXG | S_IRWXO );
+#endif
     if (fd == -1) return STATUS_INTERNAL_ERROR;
 
     if (ftruncate( fd, info->map_size )) goto err;
diff --git a/server/fsync.c b/server/fsync.c
index dc50aa0..e6abc28 100644
--- a/server/fsync.c
+++ b/server/fsync.c
@@ -53,12 +53,14 @@
 
 int do_fsync(void)
 {
-#ifdef __linux__
+#if defined(__linux__) && !defined(__ANDROID__)
     static int do_fsync_cached = -1;
 
     if (do_fsync_cached == -1)
     {
+        #if defined(__linux__) && !defined(__ANDROID__)
         syscall( __NR_futex_waitv, 0, 0, 0, 0, 0);
+        #endif
         do_fsync_cached = getenv("WINEFSYNC") && atoi(getenv("WINEFSYNC")) && errno != ENOSYS;
     }
 
@@ -85,8 +87,10 @@ static uint32_t shm_idx_free_search_start_hint;
 static void shm_cleanup(void)
 {
     close( shm_fd );
+#if defined(__linux__) && !defined(__ANDROID__)
     if (shm_unlink( shm_name ) == -1)
         perror( "shm_unlink" );
+#endif
 }
 
 void fsync_init(void)
@@ -97,14 +101,16 @@ void fsync_init(void)
         fatal_error( "cannot stat config dir\n" );
 
     if (st.st_ino != (unsigned long)st.st_ino)
-        sprintf( shm_name, "/wine-%lx%08lx-fsync", (unsigned long)((unsigned long long)st.st_ino >> 32), (unsigned long)st.st_ino );
+        sprintf( shm_name, "/data/data/com.termux/files/usr/tmp/wine-%lx%08lx-fsync", (unsigned long)((unsigned long long)st.st_ino >> 32), (unsigned long)st.st_ino );
     else
-        sprintf( shm_name, "/wine-%lx-fsync", (unsigned long)st.st_ino );
-
+        sprintf( shm_name, "/data/data/com.termux/files/usr/tmp/wine-%lx-fsync", (unsigned long)st.st_ino );
+#if defined(__linux__) && !defined(__ANDROID__)
     if (!shm_unlink( shm_name ))
         fprintf( stderr, "fsync: warning: a previous shm file %s was not properly removed\n", shm_name );
-
+#endif
+#if defined(__linux__) && !defined(__ANDROID__)
     shm_fd = shm_open( shm_name, O_RDWR | O_CREAT | O_EXCL, 0644 );
+#endif
     if (shm_fd == -1)
         perror( "shm_open" );
 
